sashimi-webapp/src/database/storage.js

```diff
@@ -0,0 +1,290 @@
+/**
+ *
+ *  CS3283/4 storage.js
+ *  This class acts as a facade for other developers to access to the database.
+ *  The implementation is a non-SQL local storage to support the WebApp.
+ *
+ */
+
+import constants from './constants';
+
+import entitiesCreator from './create/entitiesCreator';
+
+import query from './retrieve/query';
+
+import dataModifier from './data-modifier/dataModifier';
+
+import exceptions from './exceptions';
+
+let databaseName = constants.INDEXEDDB_NAME;
+
+// dummy function to initialize createTables promises
+function initCreateTable() {
+  if (typeof Promise === 'function') {
+    return new Promise((resolve, reject) => {
+      resolve(true);
+    });
+  } else {
+    throw new exceptions.PromiseFunctionNotDefined();
+  }
+}
+
+function createUserTable(isUserTableFirstInitialize) {
```

Comment:
```
function createUserTable(isUserTableFirstInitialize) {
  ...
function createOrganizationTable(
  ...
function createFolderTable(
  ...
function createFileManagerTable(
  ...
```

Looks like these code shared very similar structure and code base. 

Note that in JavaScript you can pass function around as an object.
So you may be interested to explore some way to not repeat these codes. Perhaps this is something related to functional programming.

It is however not critical for now, just thought you would be interested.